File: 01.backend/src/main/java/com/fallapp/model/Evento.java
package com.fallapp.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;

/**
 * Entidad Evento
 * Tabla: eventos
 */
@Entity
@Table(name = "eventos")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Evento {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_evento")
    private Long idEvento;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_falla", nullable = false)
    @NotNull(message = "La falla es obligatoria")
    private Falla falla;

    @Enumerated(EnumType.STRING)
    @Column(name = "tipo", nullable = false, length = 20, columnDefinition = "tipo_evento")
    private TipoEvento tipo;

    @Column(name = "nombre", nullable = false, length = 255)
    @NotBlank(message = "El nombre es obligatorio")
    private String nombre;

    @Column(name = "descripcion", columnDefinition = "TEXT")
    private String descripcion;

    @Column(name = "fecha_evento", nullable = false)
    @NotNull(message = "La fecha del evento es obligatoria")
    private LocalDateTime fechaEvento;

    @Column(name = "ubicacion", length = 255)
    private String ubicacion;

    @Column(name = "direccion", length = 255)
    private String direccion;

    @Column(name = "imagen_nombre", length = 255)
    private String imagenNombre;

    @Column(name = "participantes_estimado")
    private Integer participantesEstimado;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "creado_por")
    private Usuario creadoPor;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "actualizado_por")
    private Usuario actualizadoPor;

    @CreationTimestamp
    @Column(name = "fecha_creacion", nullable = false, updatable = false)
    private LocalDateTime creadoEn;

    @UpdateTimestamp
    @Column(name = "actualizado_en")
    private LocalDateTime actualizadoEn;

    /**
     * Enum para tipos de evento
     */
    public enum TipoEvento {
        planta,
        crema,
        ofrenda,
        infantil,
        concierto,
        exposicion,
        encuentro,
        cena,
        teatro,
        otro
    }
}

File: 03.mobile/app/src/main/java/com/fallapp/features/eventos/domain/model/Evento.kt
package com.fallapp.features.eventos.domain.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.LocalDateTime
import java.time.format.DateTimeFormatter

/**
 * Modelo de dominio para eventos de falla.
 */
@Serializable
data class Evento(
    @SerialName("id_evento")
    val idEvento: Long,

    @SerialName("id_falla")
    val idFalla: Long,

    @SerialName("nombreFalla")
    val nombreFalla: String,

    @SerialName("tipo")
    val tipo: String,

    @SerialName("nombre")
    val nombre: String,

    @SerialName("descripcion")
    val descripcion: String? = null,

    @SerialName("fecha_evento")
    val fechaEventoString: String,

    @SerialName("ubicacion")
    val ubicacion: String? = null,

    @SerialName("participantesEstimado")
    val participantesEstimado: Int? = null,

    @SerialName("imagenNombre")
    val imagen: String? = null,

    @SerialName("creado_por")
    val creadoPor: Long? = null,

    @SerialName("fecha_creacion")
    val fechaCreacion: String? = null
) {
    val fechaEvento: LocalDateTime
        get() {
            return try {
                val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss")
                LocalDateTime.parse(fechaEventoString, formatter)
            } catch (e: Exception) {
                LocalDateTime.now()
            }
        }
}

